### PR TITLE
Dependency Version Update

### PR DESCRIPTION
**Pull request by zozitak**
Hello,

I solved the same problem as in https://github.com/r6security/phoenix/pull/7 
but completly in go. So basicly we shouldnt had to lose any functionality, 
but please @davidkornel, read it through.

### CODE REVIEW

File: go.mod
Comment (davidkornel) [inline go.mod:1]: As of now, 1.23 is released. It may have been worth upgrading to it. WDYT? Not much of a difference so it may not be worth the extra work now.

File: deploy/manifests/time-based-trigger-demo-amtd.yaml
Comment (davidkornel) [inline deploy/manifests/time-based-trigger-demo-amtd.yaml:1]: Does this have to do anything with the upgrade or did you just leave it in?  Just an idea, maybe we should create a separate directory for samples/demos. The common directory might be confusing if someone wants to deploy Phoenix from raw manifests.

**Review by davidkornel — COMMENTED** [review]
LGTM, i checked the side-by-side with previous PR. I just noticed there was no  `api/v1beta1/adaptivemovingtargetdefense_types.go` file included in it. This might have caused the issue/confusion. :smile: 

As some code changed we should build and push a new image version (`v0.3.0`) when this gets merged. We host the images in the [GH container registry](https://github.com/r6security/phoenix/pkgs/container/phoenix%2Famtd-operator) so it should be fairly ease to do so. Can you do it @zozitak? I can help if you are not familiar with it.

File: go.mod
Comment (zozitak) [reply go.mod:1]: Yes, i see what you mean. I had only one reason for 1.22. We have dependency libraries in v0.* and i didnt want us to debug their code. With more time, more issues come out, and that is good for us.

tldr; i used the latest kinda stable release

File: deploy/manifests/time-based-trigger-demo-amtd.yaml
Comment (zozitak) [reply deploy/manifests/time-based-trigger-demo-amtd.yaml:1]: I support the idea to create a samples/demos folder.
I propose a Phoenix with settable verbosity for easier debuging. wdyt? 

This change is intentional. We may revert it, when time-based-trigger is updated too.

**Comment by zozitak** [comment]
> LGTM, i checked the side-by-side with previous PR. I just noticed there was no `api/v1beta1/adaptivemovingtargetdefense_types.go` file included in it. This might have caused the issue/confusion. 😄

sy about that. next time i will highlight the relative changes, if that happens.
 
> As some code changed we should build and push a new image version (`v0.3.0`) when this gets merged. We host the images in the [GH container registry](https://github.com/r6security/phoenix/pkgs/container/phoenix%2Famtd-operator) so it should be fairly ease to do so. Can you do it @zozitak? I can help if you are not familiar with it.

Thanks @davidkornel, i would like to live with your offering, but in the case of suit test building for phoenix. It would be a great help. 
About the Docker Image Registry, @mariusrugan showed some example scripts, and i could push for my registry. Than i've got rights for Phoenx, so i can build and push v0.3.0.
